### PR TITLE
Bunch of minor fixes

### PR DESCRIPTION
**Pull request by erikd**
Biggest change is the .travis.yml file which tests with the last 4 major releases of GHC.

### CODE REVIEW

**Comment by simonmar** [comment]
Thanks!

File: src/ProduceCode.lhs [hunk 1]
```diff
@@ -1319,7 +1320,7 @@ slot is free or not.
 > hexChars acts = concat (map hexChar acts)
 
 > hexChar :: Int -> String
-> hexChar i | i < 0 = hexChar (i + 2^16)
+> hexChar i | i < 0 = hexChar (i + 65536)
```
Comment (int-index) [inline src/ProduceCode.lhs]: How's this an improvement?

Comment (erikd) [inline src/ProduceCode.lhs]: It prevents a compiler warning (type defaulting due to `^`). Most software engineers should recognise `65536` as `1 << 17` or `2^16`.

Comment (PhilThomas) [inline src/ProduceCode.lhs]: Yes, one or the other, but not both ;-)

**Pull request merged**